src/utils/export_mapping.py

```diff
@@ -0,0 +1,124 @@
+        valid pixels.
+
+    """
+    start_time = multispectral_image.get('collectionStartTime')
+    multispectral_date = ee.Date(start_time)
+    multispectral_geometry = multispectral_image.geometry()
+    multispectral_mask = multispectral_image.mask()
+    multispectral_mask = multispectral_mask.reduce(ee.Reducer.anyNonZero())
+
+    tropomi_images = tropomi.get_bands(multispectral_date,
+                                       multispectral_geometry)
+    road_bands = road.get_bands(multispectral_geometry)
+    latlon = ee.Image.pixelLonLat().updateMask(multispectral_mask)
+    latlon = latlon.clipToBoundsAndScale(multispectral_geometry,
+                                         scale=scale)
+
+    def add_bands(tropomi_image):
+        """Match bands from different imageries."""
+        tropomi_date = tropomi_image.date()
+        tropomi_mask = tropomi_image.mask().reduce(ee.Reducer.anyNonZero())
+
+        wind_bands = wind.get_bands(tropomi_date, multispectral_geometry)
+        dsm_bands = dsm.get_bands(tropomi_date,
+                                  multispectral_geometry,
+                                  scale=scale)
+
+        total_mask = multispectral_mask.addBands(tropomi_mask)
+        total_mask = total_mask.reduce(ee.Reducer.allNonZero())
+
+        date_bands = add_day_bands(tropomi_date, total_mask)
+        date_bands = date_bands.clipToBoundsAndScale(multispectral_geometry,
+                                                     scale=scale)
+
+        convolved_mask = total_mask.convolve(vertical_kernel)
+        convolved_mask = convolved_mask.convolve(horizontal_kernel)
+        valid_neighborhood = convolved_mask.eq(1).unmask(0, False)
+        valid_neighborhood = valid_neighborhood.rename(['valid'])
+
+        valid_pixels = valid_neighborhood.reduceRegion(ee.Reducer.sum(),
+                                                       multispectral_geometry,
+                                                       scale)
+        combined_bands = ee.Image.cat([multispectral_image, tropomi_image,
+                                       dsm_bands, wind_bands, road_bands,
+                                       date_bands, latlon,
+                                       valid_neighborhood])
+        combined_bands = combined_bands.updateMask(total_mask)
+        combined_bands = combined_bands.select(bands)
+
+        num_valid_pixels = ee.Number(valid_pixels.get('valid'))
+
+        return ee.Algorithms.If(num_valid_pixels.gt(num_samples),
+                                combined_bands,
+                                None)
+    valid_stacked_bands = tropomi_images.map(add_bands, opt_dropNulls=True)
+    return valid_stacked_bands
+
+
+def date_band(date, unit, interval, name):
+    """Return an Image with relative value of unit in interval."""
+    return ee.Image.constant(date.getRelative(unit, interval)).rename(name)
+
+
+def add_day_bands(date, mask):
```

Comment:
Should this be renamed to add_date_bands? Or add_time_interval_bands? Since we are adding more than just days?

Reply:
That makes sense